src/balancerd/src/lib.rs

```diff
@@ -568,6 +571,12 @@ impl PgwireBalancer {
             metrics.tenant_pgwire_sni_count(tenant, has_sni).inc();
         }
 
+        if LOG_PGWIRE_CONNECTION_STATUS.get(&configs) {
+            info!(%conn.uuid, "starting new pgwire connection in balancer");
+        }
+        let prev = params.insert(CONN_UUID_KEY.to_string(), conn.uuid.to_string());
+        assert_none!(prev);
```

Comment:
Won't this panic the balancer process in production if someone starts a pgwire session with this parameter set? Maybe extremely unusual but seems better to send back an error gracefully rather than panic.

Reply:
Good call, updated.